modulemd/tests/test-modulemd-moduleindex.c

```diff
@@ -1186,6 +1186,116 @@ test_module_index_read_compressed (void)
 }
 
 
+static void
+test_module_index_read_def_dir (void)
+{
+  g_autoptr (ModulemdModuleIndex) idx = modulemd_module_index_new ();
+  g_autoptr (GError) error = NULL;
+  g_autofree gchar *path =
+    g_build_path ("/", g_getenv ("TEST_DATA_PATH"), "defaults", NULL);
+  g_auto (GStrv) module_names = NULL;
+  g_autoptr (GHashTable) defaultdict = NULL;
+  gsize name_len, i, j;
+
+  g_assert_nonnull (idx);
+
+  /* First verify that it works without overrides */
+  g_assert_true (modulemd_module_index_update_from_defaults_directory (
+    idx, path, TRUE, NULL, &error));
+  g_assert_no_error (error);
+
+  /* There should be three modules in the index now:
+   * - meson
+   * - ninja
+   * - nodejs
+   */
+  module_names = modulemd_module_index_get_module_names_as_strv (idx);
+  g_assert_nonnull (module_names);
+
+  for (name_len = 0; module_names[name_len]; name_len++)
+    ;
+  g_assert_cmpint (name_len, ==, 3);
+
+  const gchar *names[] = { "meson", "ninja", "nodejs", NULL };
+  for (i = 0; names[i]; i++)
+    {
+      for (j = 0; module_names[j]; j++)
+        {
+          if (g_str_equal (names[i], module_names[j]))
+            break;
+        }
+      g_assert_cmpint (j, <=, name_len);
+    }
```

Comment:
Wouldn't it be much simpler to do the above using `g_strv_equal()`? `modulemd_module_index_get_module_names_as_strv()` already returns the names sorted.

Reply:
Of course it would. I just completely forgot that `modulemd_module_index_get_module_names_as_strv()` guarantees the sorting. Thanks, that'll simplify things quite a bit.

I had written the Python test first and then basically copied it to the C tests. That was my ham-handed attempt at recreating the `in` keyword...

Reply:
Hmm, actually I think I'm going to use `g_strv_contains()` instead, because I'd like it to be more visible. But that's still quite a lot cleaner to read than the current implementation.